storage/mysql/queue.go

```diff
@@ -47,13 +47,70 @@ func (m *MySQLStorage) EnqueueCommand(ctx context.Context, ids []string, cmd *md
 	return nil, tx.Commit()
 }
 
+func (s *MySQLStorage) deleteCommand(ctx context.Context, tx *sql.Tx, id, uuid string) error {
+	// delete command result (i.e. NotNows) and this queued command
+	_, err := tx.ExecContext(
+		ctx, `
+DELETE
+    q, r
+FROM
+    enrollment_queue AS q
+    LEFT JOIN command_results AS r
+        ON q.command_uuid = r.command_uuid AND r.id = q.id
+WHERE
+    q.id = ? AND q.command_uuid = ?;
+`,
+		id, uuid,
+	)
+	if err != nil {
+		return err
+	}
+	// now delete the actual command if no enrollments have it queued
+	// nor are there any results for it.
+	_, err = tx.ExecContext(
+		ctx, `
+DELETE
+    c
+FROM
+    commands AS c
+    LEFT JOIN enrollment_queue AS q
+        ON q.command_uuid = c.command_uuid
+    LEFT JOIN command_results AS r
+        ON r.command_uuid = c.command_uuid
+WHERE
+    c.command_uuid = ? AND
+    q.command_uuid IS NULL AND
+    r.command_uuid IS NULL;
```

Comment:
"Delete from 'commands' where the command_uuid matches the provided one, and there is no matching record in the enrollment queue, and there is no matching record in the results table."

After you have a delete that translates to:
"Delete from the enrollment queue and the results table where the id in the queue matches the provided value and the command uuid matches the provided value.  It's ok if there is nothing in the results table"

So, combined:
"For command_uuid=?, After I delete everything associated with id=? clean up the commands entry if it refers to nothing".

And we're relying on command_uuid being unique. Seems good?

Reply:
I'm a bit worried on the multi table delete used here again, due to the intricacies of its failure/ordering.
when you could always do some classic SQL command like
```sql
DELETE  c
FROM commands
WHERE NOT EXISTS (SELECT * from enrollment_queue WHERE command_uuid = c.command_uuid)
AND   NOT EXISTS (SELECT * from command_results  WHERE command_uuid = c.command_uuid)
```
but since we're only deleting one row here it probably won't matter :)